### ISSUE TITLE
Bash takes over my cpu

### ISSUE THREAD

**Issue by TheTrveAnthony** [awesomeWM/awesome#3340]
Hi,

I started building my awesome config a few weeks ago, and when I implemented my CPU widget, I noticed that cpu usage was abnormally high and that it was due to bash and a script running in background. I got rid of the script (a matrix script) but bash keeps coming back, every 2 or three minutes and it gets extremly annoying to type killall bash every 30 seconds (I use fish shell btw). Also I have the plasma de installed and it never happens when I use it so it "comes" from awesome. Does anyone have the same issue ? Any idea where it could come from ?
I also changed all the bash commands in my config into sh (dash in my case) commands, but it changed nothing.

Output of `awesome --version`:
```

awesome v4.3 (Too long)
 • Compiled against Lua 5.3.5 (running with Lua 5.3)
 • D-Bus support: ✔
 • execinfo support: ✔
 • xcb-randr version: 1.6
 • LGI version: 0.9.2
```

Thank you and have a nice evening

**Comment by Aire-One** [comment]
Since you mentioned your issue is caused by "implementation of a cpu widget with sh invocation", I guess you used `awful.widget.watch`.

The best hint I can tell right now is to slowdown the timeout and use an "optimized" sh script. Here are two recent similar issues you should check :

- https://github.com/streetturtle/awesome-wm-widgets/pull/246
- https://www.reddit.com/r/awesomewm/comments/mycecj/increasing_memory_usage_with_awfulwidgetwatch/?utm_medium=android_app&utm_source=share

**Comment by TheTrveAnthony** [comment]
Thank you for your reply,

In deed, I use `awful.widget.watch` in  6 different widgets for the moment (I didn't implement everything I want yet so it might increase) and some of them update every second. 
But that's not what caused the problem, I found out thanks to bashtop (pretty ironic, right ?) that it comes from a couple scripts (shell color scripts by distrotube) that run an infinite loop (just like the matrix script). I moved them to another directory and everything is fine now.

On CPU widget: I actually don't use the graph widget, I use `wibox.container.arcchart` instead (with ram as well) to get this : 

![cpuram](https://user-images.githubusercontent.com/40374762/116482888-57612380-a886-11eb-9eda-026b3dfba200.png)

Thanks again for your reply and my apologies for doubting this great wm, I hope my next post will be in the screenshots topic !!

Take care
